### PR TITLE
Adds filters to dashboard (on status or user)

### PR DESCRIPTION
**Pull request by jiwalker-usgs**

### CODE REVIEW

File: gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/service/BaseProcessServlet.java [hunk 1]
```diff
@@ -83,21 +82,29 @@ protected final Connection getConnection() throws SQLException {
...
 
 	/**
 	 *
 	 * @param limit the max number of results to return
 	 * @param offset which row of the query results to start returning at
+	 * @param params parameter map for querying data
 	 * @return a list of ExecuteRequest request ids
 	 * @throws SQLException
 	 */
-	protected final List<String> getRequestIds(int limit, int offset) throws SQLException {
```
Comment (isuftin) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/service/BaseProcessServlet.java]: Will removing this signature break upstream API?

Comment (jiwalker-usgs) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/service/BaseProcessServlet.java]: This is the only place it is used, needed a way to add optional params for filtering

Comment (jiwalker-usgs) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/service/BaseProcessServlet.java]: I could add another method if changing api here is a problem

File: gdp-process-wps/src/main/webapp/js/dashboard.js [hunk 2]
```diff
@@ -99,4 +99,15 @@ Date.prototype.timeNow = function () {
 
 function zeroPadSingleDigit (value) {
 	return value < 10 ? "0" + value : value;
+}
+
+function listUrl(page, status, hash) {
```
Comment (isuftin) [inline gdp-process-wps/src/main/webapp/js/dashboard.js]: Any concerns here with escaping?

File: gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/service/BaseProcessServlet.java (hunk 1 above)
Comment (isuftin) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/service/BaseProcessServlet.java]: If nothing else is using it, don't bother. Otherwise, yeah just re-add getRequestIds(DEFAULT_LIMIT, DEFAULT_OFFSET) and have it feed the new method with a blank treemap

File: gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/service/BaseProcessServlet.java [hunk 3]
```diff
@@ -121,4 +128,25 @@ protected final String getIdentifier(String xml) throws JAXBException, IOExcepti
 		String identifier = execute.getIdentifier().getValue();
 		return identifier.substring(identifier.lastIndexOf(".") + 1);
 	}
+	
+	private String requestQueryBuilder(Map<String, String[]> params) {
+		StringBuilder builder = new StringBuilder();
+		builder.append("SELECT")
+				.append(" r.request_id, r.wps_algorithm_identifier, r.status, r.creation_time, r.start_time, r.end_time")
+				.append(" FROM response r, request_metadata m");
+		if (params.containsKey("hash") || params.containsKey("status")) {
```
Comment (isuftin) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/service/BaseProcessServlet.java]: Is it possible for "hash" or "status" to come in blank or nonsensical and possibly break the function/inject awful?

I guess maybe a validity test here?

File: gdp-process-wps/src/main/webapp/dashboard.jsp [hunk 4]
```diff
@@ -9,6 +9,10 @@
         <link rel="stylesheet" type="text/css" href="css/dashboard.css" />
         <jsp:include page="scripts.jsp" />
         <script src="js/dashboard.js"></script>
+        <script>
+            var status = '<%= (request.getParameter("status")==null) ? "" : request.getParameter("status")%>';
```
Comment (isuftin) [inline gdp-process-wps/src/main/webapp/dashboard.jsp]: Might want to do a trim fir the case where there is something coming in?

Comment (jiwalker-usgs) [inline gdp-process-wps/src/main/webapp/dashboard.jsp]: I think failing by returning nothing when not the null case is fine.  This dashboard is currently not meant to handle all edge cases.

File: gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/service/BaseProcessServlet.java (hunk 3 above)
Comment (jiwalker-usgs) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/service/BaseProcessServlet.java]: There is probably some validity tests that could go on here, but injection shouldn't be much of an issue, and breaking should just result in an empty result set.

**Comment by isuftin** [comment]
@jiwalker-usgs Beyond few edge cases I've laid out, this looks good to go.
